File: src/dart/lib/src/message_types.dart
/// DataChannel message-type vocabulary, exactly matching the JavaScript
/// implementation's `MESSAGE_TYPES` (src/js/src/message-types.js).
///
/// String values must stay byte-compatible across the Dart and JS
/// implementations for wire-level interop.
abstract final class MeridianMessageTypes {
  // RTT measurement.
  static const String ping = 'ping';
  static const String pong = 'pong';

  // Gossip.
  static const String gossip = 'gossip';

  // Query routing.
  static const String queryForward = 'query_forward';
  static const String leaderQueryForward = 'leader_query_forward';
  static const String constraintQueryForward = 'constraint_query_forward';
  static const String probeRequest = 'probe_request';
  static const String probeResult = 'probe_result';
  static const String probeRequestAvg = 'probe_request_avg';
  static const String probeResultAvg = 'probe_result_avg';
  static const String probeRequestConstraints = 'probe_request_constraints';
  static const String probeResultConstraints = 'probe_result_constraints';
  static const String queryResult = 'query_result';

  // Media.
  static const String mediaOffer = 'media_offer';
  static const String mediaAnswer = 'media_answer';
  static const String forwardedStream = 'forwarded_stream';
  static const String mediaClose = 'media_close';

  // Supernode / Raft.
  static const String supernodeElected = 'supernode_elected';
  static const String raftAppendEntries = 'raft_append_entries';
  static const String raftAppendEntriesResponse =
      'raft_append_entries_response';
  static const String raftRequestVote = 'raft_request_vote';
  static const String raftRequestVoteResponse = 'raft_request_vote_response';

  // Peer management.
  static const String peerLeaving = 'peer_leaving';
  static const String peerStatus = 'peer_status';

  /// Every message type in the shared wire vocabulary.
  static const Set<String> all = {
    ping,
    pong,
    gossip,
    queryForward,
    leaderQueryForward,
    constraintQueryForward,
    probeRequest,
    probeResult,
    probeRequestAvg,
    probeResultAvg,
    probeRequestConstraints,
    probeResultConstraints,
    queryResult,
    mediaOffer,
    mediaAnswer,
    forwardedStream,
    mediaClose,
    supernodeElected,
    raftAppendEntries,
    raftAppendEntriesResponse,
    raftRequestVote,
    raftRequestVoteResponse,
    peerLeaving,
    peerStatus,
  };
}

File: src/js/src/message-types.js
// DataChannel message types, exactly per spec §7.1. Field names must stay
// byte-compatible with the Dart implementation.
export const MESSAGE_TYPES = {
  // RTT measurement
  PING: 'ping',
  PONG: 'pong',

  // Gossip
  GOSSIP: 'gossip',

  // Query routing
  QUERY_FORWARD: 'query_forward',
  LEADER_QUERY_FORWARD: 'leader_query_forward',
  CONSTRAINT_QUERY_FORWARD: 'constraint_query_forward',
  PROBE_REQUEST: 'probe_request',
  PROBE_RESULT: 'probe_result',
  PROBE_REQUEST_AVG: 'probe_request_avg',
  PROBE_RESULT_AVG: 'probe_result_avg',
  PROBE_REQUEST_CONSTRAINTS: 'probe_request_constraints',
  PROBE_RESULT_CONSTRAINTS: 'probe_result_constraints',
  QUERY_RESULT: 'query_result',

  // Media
  MEDIA_OFFER: 'media_offer',
  MEDIA_ANSWER: 'media_answer',
  FORWARDED_STREAM: 'forwarded_stream',
  MEDIA_CLOSE: 'media_close',

  // Supernode / Raft
  SUPERNODE_ELECTED: 'supernode_elected',
  RAFT_APPEND_ENTRIES: 'raft_append_entries',
  RAFT_APPEND_ENTRIES_RESPONSE: 'raft_append_entries_response',
  RAFT_REQUEST_VOTE: 'raft_request_vote',
  RAFT_REQUEST_VOTE_RESPONSE: 'raft_request_vote_response',

  // Peer management
  PEER_LEAVING: 'peer_leaving',
  PEER_STATUS: 'peer_status'
};